src/Process/Pool.php

```diff
@@ -63,24 +56,17 @@ protected function _childExitSignal(InformationInterface $information): PoolInte
 
     public function freeChildProcess(int $childProcessId): PoolInterface
     {
-        try {
-            $this->_getProcessSignal()->incrementWaitCount();
-            if (isset($this->_childProcesses[$childProcessId])) {
-                if ($this->_childProcesses[$childProcessId] instanceof ProcessInterface) {
-                    $typeCode = $this->_childProcesses[$childProcessId]->getTypeCode();
-                    $this->_getLogger()->info("Freeing child process related to process[$childProcessId][$typeCode].");
-                    unset($this->_childProcesses[$childProcessId]);
-                } else {
-                    $message = "Process associated to process[$childProcessId] is not an expected type.";
-                    throw new \UnexpectedValueException($message);
-                }
+        if (isset($this->_childProcesses[$childProcessId])) {
+            if ($this->_childProcesses[$childProcessId] instanceof ProcessInterface) {
+                $typeCode = $this->_childProcesses[$childProcessId]->getTypeCode();
+                $this->_getLogger()->debug("Freeing child process related to process[$childProcessId][$typeCode].");
+                unset($this->_childProcesses[$childProcessId]);
             } else {
-                throw new \LogicException("Process associated to process[$childProcessId] is not in the process pool.");
+                $message = "Process associated to process[$childProcessId] is not an expected type.";
+                throw new \UnexpectedValueException($message);
             }
-            $this->_getProcessSignal()->decrementWaitCount();
-        } catch (\Throwable $throwable) {
-            $this->_getProcessSignal()->decrementWaitCount();
-            throw $throwable;
+        } else {
+            throw new \LogicException("Process associated to process[$childProcessId] is not in the process pool.");
```

Comment:
@rhift this is the situation you said we shouldn't ever run into again, right? (just curious, the exception should definitely be left in if it's exceptional behavior)

Reply:
@mucha55 - yeah, that is definitely a huge problem if it gets there, since it is checked at [43](https://github.com/neighborhoods/kojo/blob/a161cab3ef1063632f0b1619107545594a68ea33/src/Process/Pool.php#L43) and the message that we actually see in logs is emitted at [51](https://github.com/neighborhoods/kojo/blob/a161cab3ef1063632f0b1619107545594a68ea33/src/Process/Pool.php#L51)